### ISSUE TITLE
all_gather before loss calculation & backward pass

### ISSUE THREAD

**Issue by vanIvan** [pytorch/xla#5292]
## ❓ Questions and Help
Hello! I have one question, concerning how training will work (gradient updates / synchronization), in case when I do call `all_gather` on outputs of my network, and feed all embeddings from all cores/train processes into loss, and the do common training loop steps like `loss.backward()` and `xm.optimizer_step(optimizer)`? Is it good or bad practice?

**Comment by vanIvan** [comment]
Yes, seems like more pytorch question, thank you for answer!
